common/docker/configuration.go

```diff
@@ -0,0 +1,335 @@
+				Timeout:  3 * time.Second, //nolint:gomnd
+				Retries:  5,               //nolint:gomnd
+			},
+		},
+		HostConfig: &container.HostConfig{
+			PortBindings: nat.PortMap{
+				"26257/tcp": []nat.PortBinding{{HostPort: "26257"}},
+				"8080/tcp":  []nat.PortBinding{{HostPort: "8080"}},
+			},
+			RestartPolicy: container.RestartPolicy{Name: "unless-stopped"},
+			Mounts: []mount.Mount{{Type: mount.TypeVolume, Source: cfg.DockerEnv["CARDINAL_NAMESPACE"],
+				Target: "/var/lib/cockroach"}},
+			NetworkMode: container.NetworkMode(cfg.DockerEnv["CARDINAL_NAMESPACE"]),
+		},
+	}
+}
+
+func CelestiaDevNet(cfg *config.Config) Config {
+	return Config{
+		Name: getCelestiaDevNetContainerName(cfg),
+		Config: &container.Config{
+			Image: "ghcr.io/rollkit/local-celestia-devnet:latest",
+			ExposedPorts: nat.PortSet{
+				"26657/tcp": struct{}{},
+				"26658/tcp": struct{}{},
+				"26659/tcp": struct{}{},
+				"9090/tcp":  struct{}{},
+			},
+			Healthcheck: &container.HealthConfig{
+				Test:     []string{"CMD", "curl", "-f", "http://127.0.0.1:26659/head"},
+				Interval: 1 * time.Second,
+				Timeout:  1 * time.Second,
+				Retries:  20, //nolint:gomnd
+			},
+		},
+		HostConfig: &container.HostConfig{
+			PortBindings: nat.PortMap{
+				"26657/tcp": []nat.PortBinding{{HostPort: "26657"}},
+				"26658/tcp": []nat.PortBinding{{HostPort: "26658"}},
+				"26659/tcp": []nat.PortBinding{{HostPort: "26659"}},
+				"9090/tcp":  []nat.PortBinding{{HostPort: "9090"}},
+			},
+			RestartPolicy: container.RestartPolicy{Name: "on-failure"},
+			NetworkMode:   container.NetworkMode(cfg.DockerEnv["CARDINAL_NAMESPACE"]),
+		},
+	}
+}
+
+func EVM(cfg *config.Config) Config {
+	daBaseURL := cfg.DockerEnv["DA_BASE_URL"]
+	if daBaseURL == "" {
+		daBaseURL = "http://celestia-devnet"
+	}
+
+	daAuthToken := cfg.DockerEnv["DA_AUTH_TOKEN"]
+	if daAuthToken == "" {
+		daAuthToken = ""
+	}
+
+	faucetEnabled := cfg.DockerEnv["FAUCET_ENABLED"]
+	if faucetEnabled == "" {
+		faucetEnabled = "false"
+	}
+
+	faucetAddress := cfg.DockerEnv["FAUCET_ADDRESS"]
+	if faucetAddress == "" {
+		faucetAddress = "aa9288F88233Eb887d194fF2215Cf1776a6FEE41"
+	}
+
+	faucetAmount := cfg.DockerEnv["FAUCET_AMOUNT"]
+	if faucetAmount == "" {
+		faucetAmount = "0x56BC75E2D63100000"
+	}
+
+	baseShardRouterKey := cfg.DockerEnv["BASE_SHARD_ROUTER_KEY"]
+	if baseShardRouterKey == "" {
+		baseShardRouterKey = "abcdefghijklmnopqrstuvwxyz0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ01"
+	}
+
+	return Config{
+		Name: getEVMContainerName(cfg),
+		Config: &container.Config{
+			Image: "ghcr.io/argus-labs/world-engine-evm:1.4.1",
+			Env: []string{
+				fmt.Sprintf("DA_BASE_URL=%s", daBaseURL),
+				fmt.Sprintf("DA_AUTH_TOKEN=%s", daAuthToken),
+				fmt.Sprintf("FAUCET_ENABLED=%s", faucetEnabled),
+				fmt.Sprintf("FAUCET_ADDRESS=%s", faucetAddress),
+				fmt.Sprintf("FAUCET_AMOUNT=%s", faucetAmount),
+				fmt.Sprintf("BASE_SHARD_ROUTER_KEY=%s", baseShardRouterKey),
+			},
+			ExposedPorts: nat.PortSet{
```

Comment:
Both `ExposedPorts` and `PortBindings` seems repetitive and prone to typo. Would it be possible to create a function `getExposedPorts(ports []int)` and `getPortMap(ports []int)` respectively?

Reply:
Created the func and modified the code